jerry-core/parser/js/bc/bytecode-data.cpp

```diff
@@ -0,0 +1,865 @@
+} jerry_snapshot_bytecode_header_t;
+
+static void
+bc_add_bytecode_data (bytecode_data_header_t *bc_header_p, /**< byte-code scope data header */
+                      lit_id_hash_table *lit_id_hash_table_p, /**< (idx, block id) -> literal hash table */
+                      vm_instr_t *bytecode_p, /**< byte-code instructions array */
+                      mem_cpointer_t *declarations_p, /**< array of function / variable declarations */
+                      uint16_t func_scopes_count, /**< number of function declarations / expressions
+                                                   *   located immediately in the corresponding scope */
+                      uint16_t var_decls_count, /**< number of variable declarations immediately in the scope */
+                      bool is_strict, /**< is the scope's code strict mode code? */
+                      bool is_ref_arguments_identifier, /**< does the scope's code
+                                                        *    reference 'arguments' identifier? */
+                      bool is_ref_eval_identifier, /**< does the scope's code
+                                                    *   reference 'eval' identifier? */
+                      bool is_arguments_moved_to_regs, /**< is it function scope, for which arguments
+                                                        *   are located on registers, not in variables? */
+                      bool is_no_lex_env) /**< is lexical environment unused in the scope? */
+{
+  MEM_CP_SET_POINTER (bc_header_p->lit_id_hash_cp, lit_id_hash_table_p);
+  bc_header_p->instrs_p = bytecode_p;
+  bc_header_p->instrs_count = 0;
+  MEM_CP_SET_POINTER (bc_header_p->declarations_cp, declarations_p);
+  bc_header_p->func_scopes_count = func_scopes_count;
+  bc_header_p->var_decls_count = var_decls_count;
+  bc_header_p->next_header_cp = MEM_CP_NULL;
+
+  bc_header_p->is_strict = is_strict;
+  bc_header_p->is_ref_arguments_identifier = is_ref_arguments_identifier;
+  bc_header_p->is_ref_eval_identifier = is_ref_eval_identifier;
+  bc_header_p->is_args_moved_to_regs = is_arguments_moved_to_regs;
+  bc_header_p->is_no_lex_env = is_no_lex_env;
+} /* bc_add_bytecode_data */
+
+static void
+bc_free_bytecode_data (bytecode_data_header_t *bytecode_data_p) /**< byte-code scope data header */
+{
+  bytecode_data_header_t *next_to_handle_list_p = bytecode_data_p;
+
+  while (next_to_handle_list_p != NULL)
+  {
+    bytecode_data_header_t *bc_header_list_iter_p = next_to_handle_list_p;
+    next_to_handle_list_p = NULL;
+
+    while (bc_header_list_iter_p != NULL)
+    {
+      bytecode_data_header_t *header_p = bc_header_list_iter_p;
+
+      bc_header_list_iter_p = MEM_CP_GET_POINTER (bytecode_data_header_t, header_p->next_header_cp);
+
+      mem_cpointer_t *declarations_p = MEM_CP_GET_POINTER (mem_cpointer_t, header_p->declarations_cp);
+
+      for (uint32_t index = 0; index < header_p->func_scopes_count; index++)
```

Comment:
As was discussed in the chat, C99 supports for (int i...) way of variables declaration.